subprojects/core/src/main/java/org/gradle/configuration/ApplyScriptPluginBuildOperationType.java

```diff
@@ -33,33 +31,40 @@
     public interface Details {
 
         /**
-         * The absolute path to the file.
+         * The absolute path to the script file.
+         * Null if was not a script file.
+         * Null if uri != null.
          */
         @Nullable
         String getFile();
 
-        String getDisplayName();
-
-    }
-
-    public static class DetailsImpl implements Details {
-
-        private File file;
-        private String displayName;
+        /**
+         * The URI of the script.
+         * Null if was not applied as URI.
+         * Null if file != null.
+         */
+        @Nullable
+        String getUri();
 
-        public DetailsImpl(@Nullable File file, String displayName) {
-            this.file = file;
-            this.displayName = displayName;
-        }
+        /**
+         * The target of the script.
+         * One of "gradle", "settings", "project" or null.
+         * If null, the target is an arbitrary object.
+         */
+        @Nullable
+        String getTargetType();
 
+        /**
+         * If the target is a project, its path.
+         */
         @Nullable
-        public String getFile() {
-            return file.getAbsolutePath();
-        }
+        String getTargetPath();
 
-        public String getDisplayName() {
-            return displayName;
-        }
+        /**
+         * The build path, if the target is a known type (i.e. targetType != null)
+         */
+        @Nullable
+        String getBuildPath();
 
```

Comment:
Will plugins defined in buildSrc be represented by this type build od type or by the other pluginapplied build op type?